doc/operator-creation.md

```diff
@@ -98,9 +97,49 @@ Observable.prototype.mySimpleOperator = mySimpleOperator;
 someObservable.mySimpleOperator(x => x + '!');
 ```
 
+### Operator as a pure function
+
+If you don't want to patch the Observable prototype, you can also write the operator as a pure function that takes the input Observable as argument, instead of relying on the `this` keyword.
+
+Example implementation:
+
+```js
+function mySimpleOperator(someCallback) {
+  // notice that we return a function here
+  return function mySimpleOperatorImplementation(source) {
+    return Observable.create(subscriber => {
+      var subscription = source.subscribe(value => {
+        try {
+          subscriber.next(someCallback(value));
+        } catch(err) {
+          subscriber.error(err);
+        }
+      },
+      err => subscriber.error(err),
+      () => subscriber.complete());
+
+      return subscription;
+   });
+  }
+}
+```
+
+This can now be used with the `let()` method on the Observable:
+
+```js
+const obs = someObservable.let(mySimpleOperator(x => x + '!'));
+```
+
+## Publish your operator as a separate library
+
+We strongly recommend you to publish your custom operator as an npm package, either as a prototype function to be patched, or as a pure function to be used with `let`. See the sections above. RxJS core already has over 100 operators, and we should not add more operators unless they are absolutely essential and add functionality not possible with existing operators.
```

Comment:
Given future plans to try to move away from prototype patching, I think we should guide them to make a lettable function.

Other than this, and what was pointed out above, I think this is good. I'll try to merge as soon as the changes are in. Sorry for the delayed review.